ikarus/io/vtkwriter.hh

```diff
@@ -0,0 +1,182 @@
+                                                              Dune::Vtk::ContinuousDataCollector<GridView>>,
+                                           DC>;
+
+  // We are using a RectilinearGridWriter if structured is true
+  using UnderlyingVTKWriter = std::conditional_t<structured, Dune::Vtk::RectilinearGridWriter<GridView, DataCollector>,
+                                                 Dune::Vtk::UnstructuredGridWriter<GridView, DataCollector>>;
+
+  template <class... Args>
+  explicit Writer(std::shared_ptr<Assembler> assembler, Args... args)
+      : writer_(assembler->gridView(), args...),
+        assembler_(assembler) {}
+
+  template <class... Args>
+  Writer(std::shared_ptr<Assembler> assembler, DataCollector& dc, Args... args)
+      : writer_(dc, args...),
+        assembler_(assembler) {}
+
+  template <class... Args>
+  Writer(std::shared_ptr<Assembler> assembler, DataCollector&& dc, Args... args)
+      : writer_(std::move(dc), args...),
+        assembler_(assembler) {}
+
+  // Sets the VTK file format
+  void setFormat(Dune::Vtk::FormatTypes format) { writer_.setFormat(format); }
+
+  /// Sets the global datatype used for coordinates and other global float values
+  void setDatatype(Dune::Vtk::DataTypes datatype) { writer_.setDatatype(datatype); }
+
+  /// Sets the integer type used in binary data headers
+  void setHeadertype(Dune::Vtk::DataTypes datatype) { writer_.setHeadertype(datatype); }
+
+  auto& writer() const { return writer_; }
+  auto& writer() { return writer_; }
+
+  template <typename RF, Impl::Concepts::DataType DT>
+  void addResultFunction(RF&& resultFunction, DT /*type */) {
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<RF>(resultFunction));
+    else
+      writer_.addPointData(std::forward<RF>(resultFunction));
+  }
+
+  template <template <typename, int, int> class RT, Impl::Concepts::DataType DT>
+  requires(Concepts::ResultType<RT>)
+  void addResult(DT type) {
+    auto resFunction = makeResultVtkFunction<RT>(assembler_);
+    addResultFunction(std::move(resFunction), type);
+  }
+
+  template <Impl::Concepts::DataType DT>
+  void addAllResults(DT type) {
+    using ResultTuple = typename FEType::SupportedResultTypes;
+
+    Dune::Hybrid::forEach(ResultTuple(), [&]<typename RT>(RT i) { addResult<RT::template Rebind>(type); });
+  }
+
+  template <typename GF, Impl::Concepts::DataType DT>
+  void addGridFunction(GF&& gridFunction, const std::string& name, size_t size, DT /* type */) {
+    auto fieldInfo = Dune::Vtk::FieldInfo(name, size);
+    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
+      writer_.addCellData(std::forward<GF>(gridFunction), fieldInfo);
+    else
+      writer_.addPointData(std::forward<GF>(gridFunction), fieldInfo);
+  }
+
+  template <typename GF, typename FieldInfo, Impl::Concepts::DataType DT>
+  requires(std::same_as<FieldInfo, Dune::VTK::FieldInfo> or std::same_as<FieldInfo, Dune::Vtk::FieldInfo>)
```

Comment:
All function except `addInterpolation` and the result function stuff has almost the exact same body.

Therefore this could be refactored to:

```cpp
   template< Impl::Concepts::DataType DT,typename...Args>
  void add( DT /* type */, Args&& ... args) {
    if constexpr (std::is_same_v<DT, Impl::AsCellData>)
      writer_.addCellData(std::forward<Args>(args)....);
    else
      writer_.addPointData(std::forward<Args>(args)...);
  }
```
So there is no new functionality, but only different function names for the same functionality.

Therefore the only question is if we want to have the syntactic sugar of 


```